src/GameDataReader/Common/Refractor/V1/Files/UtilityMethodsRefractorV1ConfigFile.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace GameDataReader.Common.Refractor.V1.Files;
+
+internal class UtilityMethodsRefractorV1ConfigFile
+{
+    public string GetProcessPath(string modName)
+    {
+        var matchingProcesses = Process.GetProcessesByName(modName);
+        return matchingProcesses.Length == 1 ? matchingProcesses[0].GetMainModuleFileName(modName) : string.Empty;
```

Comment:
I know @TwitchPlaysBF2 is not a fan of ternary operators, since they are a bit harder to read than regular if statements. So I'd refactor this to a regular if statement.

Also, it is possible to launch multiple instances of Refractor games. While that is probably a total edge case, it can be handled by simply checking whether the process list contains any items (instead of checking it contains exactly one item).

Both put together might look something like this:

````c#
if (matchingProcesses.Length > 0) {
    return matchingProcesses[0].GetMainModuleFileName(modName);
}
return string.Empty;

Reply:
Before this gets touched I am wondering what this DLL function call is for. Does it have to do with the parent process or something? 

If not, and the goal is, to just read the process directory, then this would just do it, without that dll import or the `GetMainModuleFileName` method:

```csharp
public string GetProcessPath(string modName)
{
    var foundProcess = Process.GetProcessesByName(modName).FirstOrDefault();
    var filePath = foundProcess?.MainModule?.FileName;
    var folderPath = Path.GetDirectoryName(filePath);

    return folderPath ?? string.Empty;
}
```